### ISSUE TITLE
Scrolling with trackpad in two directions at once doesn't seem to work properly

### ISSUE THREAD

**Issue by craftyjon** [wxWidgets/wxWidgets#24516]
### Description

When some widget can be scrolled both horizontally and vertically, wxWidgets seems to get "locked in" to accepting only one scroll direction.  On macOS hardware with a trackpad input, it is easy to input "diagonal" scrolling, but this doesn't really work in wxWidgets, because one or the other direction (horizontal or vertical) will "win" and so diagonal scrolling of the widget can only be accomplished "Manhattan style".

This doesn't seem to be a limitation of the hardware, as other applications (e.g. HTML Canvas in Firefox/Chrome) can accept simultaneous scroll inputs on two axes.

I am not sure if this is a wxMac-specific problem or not, as the only hardware I have that can give this kind of input is a Macbook.

#### To Reproduce:

Run the wxGrid sample and try scrolling around with a trackpad that allows scrolling diagonally (i.e. two mousewheel inputs at once)

Compare behavior to, for example, some of the demos at https://konvajs.org/docs/sandbox/Canvas_Scrolling.html

### Platform and version information

- wxWidgets version  you use: 3.2.4
- wxWidgets port  you use: wxMac
- OS  and its version: macOS 14.4

**Comment by showier-drastic** [comment]
Can confirm this behaviour on wx. And to be specific, demo 3 in your example works diagonally.

**Comment by vadz** [comment]
We don't have a way of representing diagonal scrolling at `wxMouseEvent` level currently as it has `wxMouseWheelAxis` field which can take only one of 2 values: either `wxMOUSE_WHEEL_VERTICAL` or `wxMOUSE_WHEEL_HORIZONTAL`.

Ideal would probably be to add `wxMOUSE_WHEEL_MIXED` or something like this and also add the angle/direction or maybe just add `GetWheelRotationX()` and `GetWheelRotationY()` in addition to the existing `GetWheelRotation()`.

But, alternatively, and simpler, we could send 2 events for the horizontal and vertical components respectively, because I think any program would still handle diagonal scrolling as a sum of H/V scrolling, but I'm not totally sure about this.

AFAICS only Cocoa sends a single event with deltas in both directions, both MSW and GTK seem to only send H/V events, although maybe there is some other way to get trackpad scrolling input that would allow receiving both?

To summarize, it would be easy to send 2 events if both X and Y deltas in a `NSScrollWheel` are non-zero and so we should probably start with this. But sending a single event with the full information about scrolling in both directions would be more difficult, especially if we want to do it under all platforms.

**Comment by craftyjon** [comment]
I would be happy to test a simple version and see if, once integrated into KiCad, we can effectively act as if a single event with two directions were received.

**Comment by craftyjon** [comment]
Also, it seems like re. the cross-platform thing, maybe there could be a note that currently only wxOSX will send both X and Y in one event, but other platforms may support it in the future.  I am not sure about Windows, but I could see Linux adding that someday.

**Comment by vadz** [comment]
Just to be clear: would sending 2 events, one in H and one in V direction, for the same gesture be good enough for your needs or do you really need to have a single event?

**Comment by craftyjon** [comment]
I think we can make it work with 2 events as long as they come in interleaved.  Right now it seems like once a scroll is underway in one axis, input on the other axis is blocked.

**Comment by dsa-t** [comment]
On Win11, touchpad diagonal panning (after exceeding a H/V lock threshold) already works fine in KiCad 8.0 with "Trackpad" settings.
I get interleaved `wxEVT_MOUSEWHEEL` events with alternating axis.

**Comment by vadz** [comment]
Thanks, I'll try to implement the same behaviour for Mac too then.

**Comment by showier-drastic** [comment]
FYI backported to 3.2: https://gitlab.com/kicad/code/wxWidgets/-/merge_requests/7
